src/components/Language.js

```diff
@@ -1,107 +1,64 @@
-import { BrowserRouter as Link } from "react-router-dom";
-import styled, { css } from "styled-components";
-import { Java } from "@styled-icons/fa-brands/Java";
-import { Python } from "@styled-icons/simple-icons/Python";
-import Quiz from "./Quiz";
+import React from "react";
+import { Link } from "react-router-dom";
+// styling
+import { JavaIcon, PythonIcon, JSIcon } from "../styling/Icons";
+import { IconContainer } from "../styling/Containers";
+import { LanguageContainer } from "../styling/Containers";
 
-//Style
-const Container = styled.div`
-  display: inline-block;
-  border-radius: 3px;
-  padding: 0.5rem 0;
-  margin: 0.5rem 4rem;
-  background: white;
-  color: black;
-  text-align: center;
-  width: 500px;
-  height: 400px;
-`;
-
-const JavaIcon = styled(Java)`
-  margin: 0.5em 1em;
-  color: rgba(33, 182, 168, 1);
-`;
-
-const PythonIcon = styled(Python)`
-  margin: 0.5em 1em;
-  color: rgba(33, 182, 168, 1);
-`;
-
-function Language() {
-  const [langugage, setLanguage] = useState("");
-  console.log(langugage);
-
-  // sets the language in state to java
-  function handleJavaClick(e) {
-    // e.preventDefault();
-    console.log("The link was clicked.");
-    setLanguage("java");
-  }
-
-  // sets the language in state to javascript
-  function handleJSClick(e) {
-    // e.preventDefault();
-    console.log("The link was clicked.");
-    setLanguage("javascript");
-  }
-
-  // sets the language in state to python
-  function handlePythonClick(e) {
-    // e.preventDefault();
-    console.log("The link was clicked.");
-    setLanguage("python");
-  }
-
-  const JavaButton = withRouter(({ history }) => (
-    <JavaIcon
-      size="80"
-      type="button"
-      onClick={() => {
-        handleJavaClick();
-        history.push("/quiz");
-      }}
-    >
-      Java
-    </JavaIcon>
-  ));
+const Language = (props) => {
+  const JavaButton = () => (
+    <Link to="/quiz">
+      <JavaIcon
+        size="80"
+        type="button"
+        onClick={() => {
+          props.handleLanguageSelect("java");
+        }}
+      >
+        Java
+      </JavaIcon>
+    </Link>
+  );
 
-  // const JSButton = withRouter(({ history }) => (
-  //   <button
-  //     type="button"
-  //     onClick={() => {
-  //       handleJSClick();
-  //       history.push("/quiz");
-  //     }}
-  //   >
-  //     JavaScript
-  //   </button>
-  // ));
+  const JSButton = () => (
+    <Link to="/quiz">
+      <JSIcon
+        size="80"
+        type="button"
+        onClick={() => {
+          props.handleLanguageSelect("javascript");
+        }}
+      >
+        JavaScript
+      </JSIcon>
+    </Link>
+  );
 
-  const PythonButton = withRouter(({ history }) => (
-    <PythonIcon
-      size="80"
-      type="button"
-      onClick={() => {
-        handlePythonClick();
-        history.push("/quiz");
-      }}
-    >
-      Python
-    </PythonIcon>
-  ));
+  const PythonButton = () => (
+    <Link to="/quiz">
+      <PythonIcon
```

Comment:
repetition between the three *Icon components. 
All of them take `size`, `type`... maybe factor the commonalities into a LanguageIcon ?... which takes an image as prop?